### PR TITLE
5 - get reviews by id

### PR DESCRIPTION
**Pull request by MattScarthSaunders**
completion of get reviews by id including error handling

### CODE REVIEW

File: controllers/errors.controllers.js [hunk 1]
```diff
@@ -6,3 +6,19 @@ exports.catchAll = (err, req, res, next) => {
...
+
+exports.invalidId = (err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Invalid Id" });
+  } else {
+    next(err);
+  }
+};
+
+exports.idNotFound = (err, req, res, next) => {
```
Comment (callumu21) [inline controllers/errors.controllers.js:18]: This error handler would work for more than just invalid IDs so I'd recommend changing the name of the function to something more general. Essentially, this is going to manage any rejected promises you code yourself.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -58,9 +58,26 @@ describe("GET", () => {
           });
         });
     });
+    test("GET 200 /api/reviews/:review_id: should respond with a single review object of correct ID", () => {
+      return request(app)
+        .get("/api/reviews/1")
+        .expect(200)
+        .then((res) => {
+          expect(res.body.review).toEqual({
+            review_id: 1,
```
Comment (callumu21) [inline __tests__/app.test.js:67]: We know from the test database what this item should be so don't be afraid to put in more accurate information about what object we're expecting (e.g. the specific title, designer etc.)

**Review by callumu21 — APPROVED** [review]
A few comments for improving your test and error handling function, but otherwise looks good to merge! Nice work 👍

File: __tests__/app.test.js (hunk 2 above)
Comment (MattScarthSaunders) [reply __tests__/app.test.js:67]: OK cheers! I'll implement these changes and merge. Thanks for the feedback.

**Pull request merged**